Amazon.IonObjectMapper.PerformanceTest/PerformanceSuite.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.IonObjectMapper.PerformanceTest
+{
+    using System;
+    using System.Collections.Generic;
+
+    public class PerformanceSuite
+    {
+        private long baselineKey;
+        private int errorMargin;
+        private IDictionary<long, Metric> metrics;
+
+        public PerformanceSuite(long baselineKey, int errorMargin)
+        {
+            this.baselineKey = baselineKey;
+            this.errorMargin = errorMargin;
+            this.metrics = new Dictionary<long, Metric>();
+        }
+
+        public void AddMetric(long key, long runtime, long memoryUsed)
+        {
+            this.metrics.Add(key, new Metric { RuntimeTicks = runtime, MemoryUsedBytes = memoryUsed });
+        }
+
+        public bool IsLinearRuntime() 
+        {
+            long baseRuntime = this.metrics[this.baselineKey].RuntimeTicks / this.baselineKey;
+            Console.WriteLine($"Runtime baseline : {baseRuntime} ticks per entry");
+
+            long threshold = Convert.ToInt64(baseRuntime * (1 + (this.errorMargin / 100m)));
+            foreach (var kvp in this.metrics)
+            {
+                if (kvp.Value.RuntimeTicks / kvp.Key > threshold)
```

Comment:
I think it would be easier to understand if you defined a new variable `var numEntries = kvp.Key;` and then used that in place of `kvp.Key` (and likewise in `IsLinearMemoryUsed`).

Reply:
Done.